django_netjsonconfig/base/template.py

```diff
@@ -99,6 +145,10 @@ def clean(self, *args, **kwargs):
             self.auto_cert = False
         if self.type == 'vpn' and not self.config:
             self.config = self.vpn.auto_client(auto_cert=self.auto_cert)
+        super(AbstractTemplate, self).clean(*args, **kwargs)
```

Comment:
here you can add something like:

```python
if self.sharing != 'secret_key':
    self.key = None
```

you will have to add an assertion in the tests for this case